src/server/scripts/Northrend/Naxxramas/boss_sapphiron.cpp

```diff
@@ -123,7 +123,12 @@ class boss_sapphiron : public CreatureScript
                 _Reset();
 
                 if (_phase == PHASE_FLIGHT)
+                {
                     ClearIceBlock();
+                    me->SetReactState(REACT_AGGRESSIVE);
+                }
+
+                me->SetFullHealth();
```

Comment:
It doesn't reset react state. Try resetting Sapphiron (vanish etc) while he's in flight phase - proximity aggro will be disabled.